lib/routemaster/faraday_middleware/caching.rb

```diff
@@ -0,0 +1,46 @@
+require 'routemaster/cache'
+
+module Routemaster::FaradayMiddleware
+  class Caching
+
+    def initialize(app, cache = Routemaster::Cache.new)
+      @app = app
+      @cache = cache
+    end
+
+    def call(env)
+      return @app.call(env) unless env.method == :get
+
+      url, version, locale = url(env), version(env), locale(env)
```

Comment:
this has pretty appalling performance in ruby & is quite hard to read - please can we split them out?